src/declarations/stencil-public-compiler.ts

```diff
@@ -246,6 +246,17 @@ export interface StencilConfig {
   watchIgnoredRegex?: RegExp | RegExp[];
   excludeUnusedDependencies?: boolean;
   stencilCoreResolvedId?: string;
+
+  /**
+   * @deprecated will not be supported in a future version of Stencil.
+   *
+   * When `true`, any custom types used as a part of a component `@Prop` or
+   * `@Event` will be automatically exported from the generated `components.d.ts` file
+   * in the compiled output.
+   *
+   * @default true
+   */
+  autoExportCustomTypes?: boolean;
```

Comment:
From the PR summary:

> was immediately marked as deprecated as it was only added as an interim fix for the Ionic Framework team's migrations to Stencil V3

Do we have an idea as to how long this flag will exist? I.E. can we delete it before Stencil v3.0.0 is released? If the answer is no, can we be more explicit about steering folks away from this flag by renaming it something like:
```suggestion
  __ionicFrameworkInternalAutoExportCustomTypes?: boolean;
```

Reply:
@rwaskiewicz @alicewriteswrongs It won't be removed before V3. Do we want to follow the `deprecated` naming convention used for the `extras` options Alice recently worked on?

Reply:
It's that's the case, I'd like to make this _extra_ volatile 😈 Something like React's ` __SECRET_DOM_DO_NOT_USE_OR_YOU_WILL_BE_FIRED` so that we can remove this in a minor version of Stenci - hence the 'ionicFrameworkInternal' part. If anyone states removing that flag broke them we can say:
1. it was marked as internal only
2. it used `__` as a prefix (designating it as private)
3. AND it wasn't documented